python/test/unit/common/test_index_map.py

```diff
@@ -192,3 +192,45 @@ def test_create_submap_owner_change():
         np.arange(sub_imap.size_local + sub_imap.num_ghosts, dtype=np.int32)
     )
     assert np.array_equal(global_indices, np.arange(comm.rank * 2, comm.rank * 2 + 3))
+
+
+def test_sub_index_map_multiple_possible_owners():
+    """Check that creating a submap doesn't crash when an index need to change owner and
+    there are multiple possible new owners"""
+    comm = MPI.COMM_WORLD
+
+    if comm.size < 3:
+        return
```

Comment:
```suggestion
    if comm.size < 3:
        pytest.skip("Test only work with 3 or more processes")
```

Reply:
We now have a weekly CI run on macOS that uses three. 

It might be worth enabling macOS CI on all PR commits as the 3 case has sometimes shown issues where 2 did not.

Reply:
Does negate your point of course.

Reply:
> How often do we run CI with more than 2 processes?

We run on CircleCI with 3 MPI ranks.

Reply:
Another thing we could do is to use ipythonparallel to create specific parallel scenarios.
I use this in adios4dolfinx, example at:
https://github.com/jorgensd/adios4dolfinx/blob/main/tests/conftest.py#L12-L17
https://github.com/jorgensd/adios4dolfinx/blob/main/tests/test_original_checkpoint.py#L310-L314